app/src/main/java/com/amaze/filemanager/adapters/data/LayoutElementParcelable.java

```diff
@@ -50,18 +50,43 @@
 
     //same as hfile.modes but different than openmode in Main.java
     private OpenMode mode = OpenMode.FILE;
+    private Integer hashCode;
+
+    public LayoutElementParcelable(String path, String permissions, String symlink,
+                                   String size, long longSize,boolean header, String date,
+                                   boolean isDirectory, boolean useThumbs, OpenMode openMode) {
+        this(new File(path).getName(), path, permissions, symlink, size, longSize, header,
+                date, isDirectory, useThumbs, openMode);
+
+    }
+
 
     public LayoutElementParcelable(String title, String path, String permissions,
                                    String symlink, String size, long longSize, boolean header,
-                                   String date, boolean isDirectory, boolean useThumbs) {
+                                   String date, boolean isDirectory, boolean useThumbs, OpenMode openMode) {
         filetype = Icons.getTypeOfFile(path, isDirectory);
         @DrawableRes int fallbackIcon = Icons.loadMimeIcon(path, isDirectory);
-
+        this.mode = openMode;
         if(useThumbs) {
-            if (filetype == Icons.IMAGE || filetype == Icons.VIDEO || filetype == Icons.APK) {
-                this.iconData = new IconDataParcelable(IconDataParcelable.IMAGE_FROMFILE, path, fallbackIcon);
-            } else {
-                this.iconData = new IconDataParcelable(IconDataParcelable.IMAGE_RES, fallbackIcon);
+            switch (mode) {
```

Comment:
This seems incomplete, is this incomplete?

Reply:
This is complete and working as intended.